// image-upload-react/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var fa = require('react-icons/fa');
var styled = _interopDefault(require('styled-components'));

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  max-width: 100px;\n  max-height: 100px;\n  min-width: 50px;\n  min-height: 50px;\n  position: absolute;\n  top: -15px;\n  right: -15px;\n  z-index: 999999999;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  cursor: pointer;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  height: 100%;\n\n  img {\n    object-fit: fill !important;\n    width: 100%;\n    height: 100%;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  width: 100%;\n  height: 100%;\n  z-index: 999;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 50%;\n  left: 50%;\n  transform: translate(-50%, -50%);\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  opacity: 0;\n  width: 100%;\n  height: 100%;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  overflow: hidden;\n  width: 100%;\n  height: 100%;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  background-color: gold;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  position: relative;\n  z-index: 99;\n  box-shadow: 0 4px 3px rgba(0, 0, 0, 0.356);\n  margin-bottom: 0.5rem;\n  margin-right: auto;\n  margin-left: auto;\n  margin-top: 3rem;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var PhotoPicker = styled.div(_templateObject());
var PhotoPickerInputWrapper = styled.div(_templateObject2());
var PhotoPickerInput = styled.input(_templateObject3());
var PhotoPickerIcon = styled.div(_templateObject4());
var PhotoPickerImagePreviewWrapper = styled.div(_templateObject5());
var PhotoPickerImagePreviewBox = styled.div(_templateObject6());
var PhotoPickerDeleteIcon = styled.div(_templateObject7());

function ImageUpload(_ref) {
  var imageSrc = _ref.imageSrc,
      setImageSrc = _ref.setImageSrc,
      handleImageSelect = _ref.handleImageSelect,
      deleteIcon = _ref.deleteIcon,
      defaultDeleteIconSize = _ref.defaultDeleteIconSize,
      defaultDeleteIconColor = _ref.defaultDeleteIconColor,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {
    backgroundColor: 'gold',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
    zIndex: '99',
    boxShadow: '0 4px 3px rgba(0, 0, 0, 0.356)',
    marginBottom: '0.5rem',
    marginRight: 'auto',
    marginLeft: 'auto',
    marginTop: '3rem'
  } : _ref$style;
  React.useEffect(function () {
    if (!imageSrc === null) {
      throw new Error('please provide an imageSrc property!');
    } else if (!setImageSrc) {
      throw new Error('please provide an setImageSrc property!');
    } else if (!handleImageSelect) {
      throw new Error('please provide an handleImageSelect property!');
    }
  }, []);
  return /*#__PURE__*/React__default.createElement("div", {
    style: {
      width: style.width || 500,
      height: style.height || 300,
      position: 'relative',
      borderRadius: style.borderRadius
    }
  }, imageSrc ? /*#__PURE__*/React__default.createElement(PhotoPickerDeleteIcon, {
    onClick: function onClick() {
      setImageSrc('');
    }
  }, deleteIcon || /*#__PURE__*/React__default.createElement(fa.FaTrash, {
    size: defaultDeleteIconSize || 30,
    color: defaultDeleteIconColor || '#fff'
  })) : null, /*#__PURE__*/React__default.createElement(PhotoPicker, {
    style: style
  }, imageSrc ? /*#__PURE__*/React__default.createElement(PhotoPickerImagePreviewWrapper, null, /*#__PURE__*/React__default.createElement(PhotoPickerImagePreviewBox, null, /*#__PURE__*/React__default.createElement("img", {
    src: imageSrc,
    alt: "pick image"
  }))) : /*#__PURE__*/React__default.createElement(PhotoPickerIcon, null, /*#__PURE__*/React__default.createElement(fa.FaArrowCircleUp, {
    size: 70,
    color: "#fff"
  })), /*#__PURE__*/React__default.createElement(PhotoPickerInputWrapper, null, /*#__PURE__*/React__default.createElement(PhotoPickerInput, {
    type: "file",
    name: "photo",
    placeholder: "select the book photo",
    accept: "image/*",
    onChange: handleImageSelect,
    onClick: function onClick(event) {
      event.target.value = null;
    }
  }))));
}

module.exports = ImageUpload;
